resources/web/wwi/x3d_scene.js

```diff
@@ -226,23 +237,27 @@ class X3dScene { // eslint-disable-line no-unused-vars
     return node;
   }
 
-  getObjectByCustomId(object, id, context) {
+  getObjectById(id, skipBoundingObject = false, object = 'scene', context = {}) {
+    // @param 'object':
+    //     Global case: object is the root object in which to search for.
+    //     Special case to have a good default value: if object === 'scene', then the scene is used.
+    if (object === 'scene')
+      object = this.scene;
+
+    if (!object ||
+        (skipBoundingObject && typeof object.userData !== 'undefined' && object.userData.x3dType === 'Switch')) {
+      context.parent = undefined;
+      return undefined;
+    }
+
     if (Array.isArray(object)) {
-      for (let i = 0, l = object.length; i < l; i++) {
-        var o = this.getObjectByCustomId(object[i], id, context);
+      for (let i = 0; i < object.length; i++) {
```

Comment:
why did you changed the for loop instructions?
note that it is common to store the length of an array so that the code it is slightly faster.

Reply:
I tested on my Firefox, and it turns out that caching the length in a local variable is faster. Try it yourself here: https://jsperf.com/array-length-performance. I am getting this:
![image](https://user-images.githubusercontent.com/1264964/57832465-07cbdf00-77b8-11e9-9d9d-4a7c856dfbb9.png)

More info here: https://stackoverflow.com/questions/9592241/javascript-is-the-length-method-efficient

Reply:
In fact, my modification is not important; I will roll it back. I was fearing about the scope of `l` but it was correct. For consistency concern, we are using a lot more the second syntax.

Reply:
I propose that I will roll back this modification, and then create a PR to use this pattern everywhere.